scripts/make_report_tables.py

```diff
@@ -0,0 +1,585 @@
+    directions: list[str],
+    distributions: list[str],
+    atk_names: list[str],
+    atk_metric_name: str,
+):
+    # Filter runs
+    runs = [
+        run
+        for run in wandb_runs
+        if A_VULN in run.summary.keys() or B_VULN in run.summary.keys()
+    ]
+
+    # Model vulnerability keys
+    A_keys = make_vuln_keys("A")
+    B_keys = make_vuln_keys("B")
+
+    # Lists to store results in
+    A_vuln = [[] for _ in range(len(A_keys))]
+    A_success = [[] for _ in range(len(distributions) * len(atk_names))]
+    B_vuln = [[] for _ in range(len(B_keys))]
+    B_success = [[] for _ in range(len(distributions) * len(atk_names))]
+
+    # Extract model vulnerability stats and transfer success
+    for run in runs:
+        # If run is A
+        if run.name[-1] == "A":
+            # A -> B direction
+            direction = directions[0]
+            count = 0
+            for distribution in distributions:
+                for atk_name in atk_names:
+                    # Get A -> B transfer
+                    A_success[count].append(
+                        run.summary[direction][distribution][atk_name][atk_metric_name]
+                    )
+                    count += 1
+            # Get opposite model B vulnerability
+            new_run_name = run.name[0:-1] + "B"
+            new_run = [run for run in runs if run.name == new_run_name][0]
+            for i, key in enumerate(B_keys):
+                B_vuln[i].append(
+                    new_run.summary["vulnerability_B"][key][atk_metric_name]
+                )
+        # If run is B
+        if run.name[-1] == "B":
+            # B -> A direction
+            direction = directions[1]
+            count = 0
+            for distribution in distributions:
+                for atk_name in atk_names:
+                    # Get B -> A transfer
+                    B_success[count].append(
+                        run.summary[direction][distribution][atk_name][atk_metric_name]
+                    )
+                    count += 1
+            # Get opposite model A vulnerability
+            new_run_name = run.name[0:-1] + "A"
+            new_run = [run for run in runs if run.name == new_run_name][0]
+            for i, key in enumerate(A_keys):
+                A_vuln[i].append(
+                    new_run.summary["vulnerability_A"][key][atk_metric_name]
+                )
+
+    # Make correlation table
+    # Key order is mA_fga, m_B_fga, m_A_ba, m_B_ba where m is the model,
+    # second letter is the distribution, last bit is fga or boundary attack
+
+    # so for target vuln * A->B transfer success for fga on dist A, we want
+    # vuln_B and the 1st A success (AA_fga)
+
+    # we want FGA on 1st row, BA on 2nd
+    # A->B dist A, A->B dist B, B->A dist A, B->A dist B for columns
+    # Note: for A->B always vuln B, for B->A always vuln B
+    out = [
+        [
+            pearsonr(B_vuln[0], A_success[0]),
+            pearsonr(B_vuln[1], A_success[1]),
```

Comment:
Having mentally stepped through the code, I think `B_vuln[1]` is the BB vuln for **Boundary Attack**, and `A_success[1]` is the A->B transfer success for **FGA**

As I'm sure you appreciate, this is difficult, so please double check!

This could be remedied by either changing the index of `A_success` to 2 (I think?) or changing the order of the for loops in lines 175, 176.

Reply:
Have changed the loop order